src/inspector_profiler.cc

```diff
@@ -503,6 +507,215 @@ static void StopCoverage(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+void ProfileWatchdog::Start() {
+  if (state_ == ProfileWatchdogState::kInitialized) {
+    state_ = ProfileWatchdogState::kRunning;
+    SetTimeout();
+    int rc = uv_thread_create(&thread_, &ProfileWatchdog::Run, this);
+    CHECK_EQ(0, rc);
+  }
+}
+
+void ProfileWatchdog::Stop() {
+  if (state_ != ProfileWatchdogState::kClosing &&
+      state_ != ProfileWatchdogState::kClosed) {
+    state_ = ProfileWatchdogState::kClosing;
+  }
+}
+
+void ProfileWatchdog::Start(const FunctionCallbackInfo<Value>& args) {
+  ProfileWatchdog* watchdog = Unwrap<ProfileWatchdog>(args.Holder());
+  if (watchdog) {
+    watchdog->Start();
+  }
+}
+
+void ProfileWatchdog::Stop(const FunctionCallbackInfo<Value>& args) {
+  ProfileWatchdog* watchdog = Unwrap<ProfileWatchdog>(args.Holder());
+  if (watchdog) {
+    watchdog->Stop();
+  }
+}
+
+void ProfileWatchdog::Run(void* arg) {
+  ProfileWatchdog* wd = static_cast<ProfileWatchdog*>(arg);
+  uv_run(&wd->loop_, UV_RUN_DEFAULT);
+}
+
+void ProfileWatchdog::Timer(uv_timer_t* timer) {
+  ProfileWatchdog* w = ContainerOf(&ProfileWatchdog::timer_, timer);
```

Comment:
Done. But because we will call `uv_thread_join ` in ProfileWatchdog destructor, I think we should call `delete this`(ProfileWatchdog) in main thread when the state is `closing`.